### ISSUE TITLE
NaN gradient of the clamped-sqrt SDFs (_sdf_box, _sdf_cylinder) at interior sample points through compute_smoothed_eps

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#1085]
## Summary

`jax.grad` of the opt-in Kottke-smoothed permittivity (`compute_smoothed_eps`, `rfx/geometry/smoothing.py`) with respect to a **Box** shape parameter is NaN whenever a sample point lies strictly inside the box. The same gradient with respect to a **Sphere** radius is finite and agrees with a central finite difference to 1.6e-9 rel. Mechanism: the clamped-sqrt form of `_sdf_box` (and, by the same form, `_sdf_cylinder`). Found by the #833 AD re-check; pre-existing, independent of ab280a38, and not a #833 blocker.

## Witness (both JAX_ENABLE_X64 flags, on main `6d721a56` and on `3f3ed7e0` = parent of ab280a38)

Fixture: `Grid(freq_max=1e9, domain=(7,5,3) mm, dx=2.54e-4, cpml_layers=2)` (no Grid warnings), `Box lo=(1.3,1.1,0.7) mm hi=(4.7,3.9,2.3) mm`, ε 4.0; objective `f(s) = Σ eps_x over the interface voxels` with the box shifted by `s` along x.

| quantity | x64=0 | x64=1 |
|---|---|---|
| `jax.grad(f)(0.0)`, HEAD | **nan** | **nan** |
| `jax.grad(f)(0.0)`, 3f3ed7e0 | **nan** | **nan** |
| central FD, h = 1e-7 m, HEAD | −700683.5625 /m | −700273.024846183 /m |
| central FD, 3f3ed7e0 | −699462.875 /m | −700273.024846183 /m |

Direct probe of `_sdf_box` at an interior point (2.0, 2.0, 1.5) mm: `sdf = -7.0e-4` (≈ -6.99999975e-4 at x64=0), `d sdf / d shift = nan`; at an exterior point `sdf = 3.0e-4`, gradient `-1.0` (correct).

Sphere control (`Sphere((3.5, 2.5, 1.5) mm, r)`, ε 4.0, objective Σ eps_x, gradient w.r.t. r): HEAD x64=1 grad `2272071.1593872565` vs central FD (h=1e-8) `2272071.16304271` (rel 1.6e-9); x64=0 grad `2272071.75` (cross-flag 2.6e-7 rel). Finite and FD-validated, so the NaN is specific to the clamped SDFs, not to the smoothing path.

## Mechanism

```python
# rfx/geometry/smoothing.py, _sdf_box
ox = jnp.maximum(dx, 0.0); oy = ...; oz = ...
outside = jnp.sqrt(ox**2 + oy**2 + oz**2)   # interior point: sqrt(0)
```

At an interior sample point all three clamped terms are exactly 0, so `outside = sqrt(0)` and `d/dx sqrt(u)` at `u = 0` is `1/(2*sqrt(0)) = inf`; multiplied by the zero inner derivative it yields `0 * inf = nan`, which propagates through `jnp.where` into every interface voxel of the gradient. `_sdf_cylinder` has the same `jnp.sqrt(jnp.maximum(r, 0)**2 + jnp.maximum(h, 0)**2)` form and will NaN at any point that is inside on both r and h. The forward value is correct; only the derivative is undefined at the clamp corner.

## Scope

- Affected: direct AD through `compute_smoothed_eps` / `compute_inv_eps_tensor_diag` w.r.t. Box or Cylinder shape parameters (position, size).
- Not affected: the production design-AD path `kottke_inv_eps_from_occupancy` (`rfx/api/_execute.py`), which works on occupancy arrays and never evaluates these SDFs; Sphere shape gradients; forward values on every lane.
- Standard remedy is a safe norm (e.g. `jnp.sqrt(jnp.maximum(u, tiny))` or a `jnp.where`-guarded sqrt with a custom zero branch) — any fix needs a forward bit-identity witness on the uniform and NU smoothing lanes and a FD check of the Box gradient.

## References

Found during the #833 re-verification (2026-09-16); coordinate spine work in #802. Scratch scripts: session scratchpad `833/measure_item1.py` (Box witness) and `833-refute/ad.py` (Sphere control).

**Comment by bk-squared** [comment]
Confirmed and fixed in #1104.

Facts. On a `git archive` export of c6788ef7 the reported NaN reproduces at both flags, and it is not limited to the Box: on the #833 uniform fixture `jax.grad` of `Σ eps_x` over a fixed interface-voxel mask is `nan` for a Box x-shift, a Cylinder-z radius, a Cylinder-z x-shift, a Cylinder-x radius and a Cylinder-x height, while the Sphere-radius control is finite. The direct `_sdf_box` probe reproduces exactly as reported: interior (2.0, 2.0, 1.5) mm `sdf = -7.0e-4`, `d sdf/d shift = nan`; exterior (5.0, 2.0, 1.5) mm `sdf = +3.0e-4`, gradient `-1.0`. Mechanism is as the issue states — every clamped overhang is exactly 0 at an interior sample, so the reverse pass evaluates `0 * d/du sqrt(u)|_0 = 0 * inf`, and the smoother's `jnp.where` spreads that NaN to every interface voxel.

What the PR does. It replaces the bare `sqrt` in the exterior-distance term of `_sdf_box` and `_sdf_cylinder` with one shared `_exterior_norm(xp, *terms)` helper using the double-`where` safe norm, so `sqrt` is never differentiated at 0. The derivative at the origin becomes 0, which is the subgradient the finite difference already reports there. Forward values do not move: `compute_smoothed_eps`, `compute_inv_eps_tensor_diag` (dielectric and PEC branches) and `compute_smoothed_eps_nonuniform` on the Box, Sphere, Cylinder-z and Cylinder-x fixtures, uniform and graded NU, are bit-identical to c6788ef7 at both flags — max|d| = 0 on all 11 cases, raw uint views equal — and the #833 bitwise x64-invariance lock stays green.

Gradient witness after the fix, x64=1, central FD `h = 1e-8`: Box x-shift `nan` → `-700272.177766158` vs FD `-700272.1862` (rel 1.208e-08); Cylinder-z radius `nan` → `2601510.213` (2.761e-09); Cylinder-z x-shift `nan` → `324427.017` (4.045e-09); Cylinder-x radius `nan` → `3165308.583` (1.311e-10); Cylinder-x height `nan` → `717467.965` (1.567e-09). The Sphere control is unchanged bit-for-bit on both branches. At x64=0 every gradient is finite and agrees with the x64=1 value to 1.151e-05 rel (worst, Cylinder-z x-shift); that flag's own central FD is float32-cancellation-limited (worst 1.152e-03, and it gets worse as `h` shrinks — 1.105e-02 at h=1e-8, 4.592e-02 at h=1e-9), so the cross-flag comparison is what constrains the x64=0 gradient. Preflight/Grid output next to those numbers: the uniform fixture is silent; the graded NU fixture raises exactly one advisory on both branches and both flags, `dz_profile has max adjacent cell ratio 2.000`.

`tests/unit/geometry/test_smoothing_shape_gradient.py` pins finiteness, per-flag FD agreement and cross-flag agreement, with all three gates derived through `tests._gate_policy.gate_from_envelope` from those measured envelopes (1.9e-08 / 1.8e-03 / 1.8e-05), plus a bitwise forward check of `_exterior_norm` against plain `sqrt` on both backends. Run against c6788ef7 the file fails 29 of 34; the 5 that pass are the Sphere control's sub-tests.

What remains. Nothing for this issue. Two notes for the record. First, the issue's Sphere control quotes `2272071.159` without stating the radius it used, so the PR carries "Sphere unchanged" as a before/after equality on its own `r = 1.1e-3` fixture rather than by reproducing that number. Second, this fix covers the clamped-overhang norm only; the *unclamped* radial `sqrt` inside `_sdf_cylinder` and `_sdf_sphere` has the same NaN-at-zero form and would fire if a sample point landed exactly on the cylinder axis or the sphere centre. No fixture in this repo hits that today (every gradient above is finite), so it is left alone rather than changed without a witness — worth a separate issue if a design lane ever puts a sample there.

Test lane: `tests/unit/geometry/` is 347 passed / 7 skipped at `JAX_ENABLE_X64=0` and 346 passed / 1 failed at `=1`. The failure is the pre-existing `test_kottke_pec_limit.py::test_compute_inv_eps_tensor_diag_returns_float32`, red on the c6788ef7 export too (`1 failed, 17 passed` there). ruff clean.
